Cyrus-sasl fails to build on musl

## Issue description

The build fails to link `libsasl2.a` owing to missing gssapi functions:

```
/nix/store/9swpc2crk4161vxxglnjzpbjh7g4f6r0-bash-4.4-p23/bin/bash ../libtool   --mode=link gcc  -Wall -W -g -O2  -L/nix/store/zbva3r16qv8c2wj0zs5swmwagybwf0hy-openssl-1.0.2p-dev/lib -Wl,-rpath,/nix/store/zbva3r16qv8c2wj0zs5swmwagybwf0hy-openssl-1.0.2p-dev/lib  -o dbconverter-2 dbconverter-2.o ../sasldb/libsasldb.la ../lib/libsasl2.la -ldb  -lresolv -lresolv
mkdir .libs
gcc -Wall -W -g -O2 -L/nix/store/zbva3r16qv8c2wj0zs5swmwagybwf0hy-openssl-1.0.2p-dev/lib -Wl,-rpath,/nix/store/zbva3r16qv8c2wj0zs5swmwagybwf0hy-openssl-1.0.2p-dev/lib -o dbconverter-2 dbconverter-2.o ../sasldb/.libs/libsasldb.a -lresolv -lresolv ../lib/.libs/libsasl2.a -ldl -lresolv -lresolv -ldb -lresolv -lresolv
../lib/.libs/libsasl2.a(dlopen.o):(.data.rel+0x10): undefined reference to `gssapiv2_server_plug_init'
../lib/.libs/libsasl2.a(dlopen.o):(.data.rel+0x28): undefined reference to `gssapiv2_client_plug_init'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:512: dbconverter-2] Error 1
make[2]: Leaving directory '/tmp/nix-build-cyrus-sasl-2.1.26.drv-0/cyrus-sasl-2.1.26/utils'
make[1]: *** [Makefile:457: all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-cyrus-sasl-2.1.26.drv-0/cyrus-sasl-2.1.26'
make: *** [Makefile:364: all] Error 2
```

### Steps to reproduce

This nix expression fails to build:

```nix
{
  nixpkgs ? import (builtins.fetchTarball {
    url = "https://github.com/NixOS/nixpkgs/archive/eeeeacc9a61d44ada034e5eec72487fbbe10744f.tar.gz";
    sha256 = "1d9jrr7vaj0x2m0bd9rmz8g5bvnh7b34gj4dc7kbhj2w7m8z5mcf";
  }) {}
}:
nixpkgs.pkgsMusl.cyrus_sasl
```

The commit sha is current nixpkgs master and cyrus-sasl builds against glibc.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.67, NixOS, 18.03.133235.305f13d3d73 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.03.133235.305f13d3d73, nixos-unstable-18.03pre124947.33a4de2941d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

--- a/pkgs/development/libraries/cyrus-sasl/default.nix
+++ b/pkgs/development/libraries/cyrus-sasl/default.nix
@@ -36,15 +36,15 @@ stdenv.mkDerivation rec {
 
   configureFlags = [
     "--with-openssl=${openssl.dev}"
+    "--with-plugindir=${placeholder "out"}/lib/sasl2"
+    "--with-saslauthd=/run/saslauthd"
+    "--enable-login"
+    "--enable-shared"
   ] ++ lib.optional enableLdap "--with-ldap=${openldap.dev}";
 
-  # Set this variable at build-time to make sure $out can be evaluated.
-  preConfigure = ''
-    configureFlagsArray=( --with-plugindir=$out/lib/sasl2
-                          --with-saslauthd=/run/saslauthd
-                          --enable-login
-                        )
-  '';
+  # Avoid triggering regenerating using broken autoconf/libtool bits.
+  # (many distributions carry patches to remove/replace, but this works for now)
+  dontUpdateAutotoolsGnuConfigScripts = true;
 
   installFlags = lib.optional stdenv.isDarwin [ "framedir=$(out)/Library/Frameworks/SASL2.framework" ];
 
@@ -55,7 +55,7 @@ stdenv.mkDerivation rec {
   '';
 
   meta = {
-    homepage = http://cyrusimap.web.cmu.edu/;
+    homepage = https://www.cyrusimap.org/sasl;
     description = "Library for adding authentication support to connection-based protocols";
     platforms = platforms.unix;
   };

--- a/pkgs/development/libraries/cyrus-sasl/default.nix
+++ b/pkgs/development/libraries/cyrus-sasl/default.nix
@@ -44,7 +44,7 @@ stdenv.mkDerivation rec {
 
   # Avoid triggering regenerating using broken autoconf/libtool bits.
   # (many distributions carry patches to remove/replace, but this works for now)
-  dontUpdateAutotoolsGnuConfigScripts = true;
+  dontUpdateAutotoolsGnuConfigScripts = if stdenv.hostPlatform.isMusl then true else null;
 
   installFlags = lib.optional stdenv.isDarwin [ "framedir=$(out)/Library/Frameworks/SASL2.framework" ];
 

